spec/app/domain/authentication/authn_k8s/application_identity_spec.rb

```diff
@@ -0,0 +1,511 @@
+                               .and_return(pod_annotation)
+    allow(pod_annotation).to receive(:[])
+                               .with(:name)
+                               .and_return("authn-k8s/pod")
+    allow(pod_annotation).to receive(:[])
+                               .with(:value)
+                               .and_return("K8sPod")
+
+    allow(deployment_annotation).to receive(:values)
+                                      .and_return(deployment_annotation)
+    allow(deployment_annotation).to receive(:[])
+                                      .with(:name)
+                                      .and_return("authn-k8s/deployment")
+    allow(deployment_annotation).to receive(:[])
+                                      .with(:value)
+                                      .and_return("K8sDeployment")
+
+    allow(stateful_set_annotation).to receive(:values)
+                                        .and_return(stateful_set_annotation)
+    allow(stateful_set_annotation).to receive(:[])
+                                        .with(:name)
+                                        .and_return("authn-k8s/stateful_set")
+    allow(stateful_set_annotation).to receive(:[])
+                                        .with(:value)
+                                        .and_return("K8sStatefulSet")
+
+    allow(deployment_config_annotation).to receive(:values)
+                                             .and_return(deployment_config_annotation)
+    allow(deployment_config_annotation).to receive(:[])
+                                             .with(:name)
+                                             .and_return("authn-k8s/deployment_config")
+    allow(deployment_config_annotation).to receive(:[])
+                                             .with(:value)
+                                             .and_return("K8sDeploymentConfig")
+
+    allow(invalid_annotation).to receive(:values)
+                                   .and_return(invalid_annotation)
+    allow(invalid_annotation).to receive(:[])
+                                   .with(:name)
+                                   .and_return("authn-k8s/non_existing")
+  end
+
+  context "initialization" do
+    subject(:application_identity) {
+      Authentication::AuthnK8s::ApplicationIdentity.new(
+        host_id:          host_id,
+        host_annotations: host_annotations,
+        service_id:       good_service_id
+      )
+    }
+
+    context "Application identity in host id" do
+      let(:host_annotations) { [] }
+      let(:host_id) { "#{namespace}/#{k8s_resource_name}/#{k8s_resource_value}" }
+
+      context "with a valid application identity" do
+        context "when is namespace scoped" do
+          let(:k8s_resource_name) { "*" }
+          let(:k8s_resource_value) { "*" }
+
+          it "does not raise an error" do
+            expect { subject }.not_to raise_error
+          end
+        end
+
+        context "when is not namespace scoped" do
```

Comment:
shouldn't it be the inverse? when it is service account, then it does not raise an error? or are we specifying hierarchy that if we are not namespace scoped, then we can do operations when we use a service_account?

Reply:
we have a separation between cases that the application identity is namespace scoped and cases that the application identity is *not* namespace scoped. the service_account is just an example of a resource that is defined in the app id along with the namespace